src/main/java/com/ericsson/ei/rules/RulesHandler.java

```diff
@@ -45,32 +40,31 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 
 @Component
-@Scope(value = "thread", proxyMode = ScopedProxyMode.TARGET_CLASS)
+@Scope("thread")
```

Comment:
Is this causing any hanging threads? What is/was ScopedProxy.... used for?

Reply:
I'm not entirely sure. It was causing problems with the initialization of the RulesHandler. PostConstruct did not run with proxyMode but I'm not sure about what the consequence is. I need to check this more.

Reply:
proxyMode appears to start a new instance every time a method in the class is called which results in a large amount of calls to init(). Without it I get 2 calls to init() and with singleton scope I get 1 call to init().